src/com/hms_networks/americas/sc/thingworx/data/TWDataPayload.java

```diff
@@ -29,8 +33,22 @@
    * @return true if data point added
    */
   public synchronized boolean addDataPoint(DataPoint datapoint) {
+    // Get configured max number of data points per payload, if required
```

Comment:
JSON parsing exception will cause maxPayloadDataPoints to be Integer.MAX_VALUE.  Move the try/catch to getPayloadMaxDataPoints().

Reply:
Cleaned this up to be more in line with other config file calls